### PR TITLE
Integrate InstallClusterAutoscalerPostHook post hook functionality in cluster-setup workflow

### LINKED ISSUE #3436: Integrate InstallClusterAutoscalerPostHook post hook functionality in cluster-setup workflow
Integrate InstallClusterAutoscalerPostHook post hook functionality in cluster-setup workflow.

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #3436 
| License         | Apache 2.0

### What's in this PR?

- move deployment of Cluster Autoscaler as a separate activity (deploy-cluster-autoscaler) into cluster setup workflow
- remove updating Cluster Autoscaler from EKS cluster update since node pools are auto discovered
- use deploy-cluster-autoscaler in Azure PKE update workflow (pke-azure-update-cluster)

### Why?
To be able to deprecate post hook functionality.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] OpenAPI and Postman files updated (if needed)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### Test 

Tested on EKS, AKS, PKE on AWS/Azure

### CODE REVIEW

File: internal/cluster/clustersetup/workflow.go [hunk 1]
```diff
@@ -48,12 +50,18 @@ type WorkflowInput struct {
...
+}
+
 // Cluster represents a Kubernetes cluster.
 type Cluster struct {
 	ID           uint
 	UID          string
 	Name         string
 	Distribution string
+	Cloud        string
+	K8sVersion   string
```
Comment (sagikazarmark) [inline internal/cluster/clustersetup/workflow.go:64]: Are these used anywhere?

Comment (sancyx) [reply internal/cluster/clustersetup/workflow.go]: nope, removed it

**Review by sagikazarmark — APPROVED** [review]
LGTM, thanks

**Review by pregnor — APPROVED** [review]
In the future in case of jointly occurring code relocation, rearrangement and modification please consider separating these actions to their dedicated commits, because it makes the reviewer's job much less error-prone as different expectations need to be checked for different actions (respectively: verifying path, verifying ordering only code change, verifying semantically valid code change).

**Pull request merged**